### PR TITLE
implements verify method

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**:
implements verify method, at the moment this includes checking the cluster state before and after a new machine has been created.

**Which issue(s) this PR fixes** : see more #47

### CODE REVIEW

File: test/tools/verify/cmd/main.go [hunk 1]
```diff
@@ -51,19 +57,64 @@ func main() {
...
+		manifestReader := strings.NewReader(manifest)
+		manifestDecoder := yaml.NewYAMLToJSONDecoder(manifestReader)
+		err := manifestDecoder.Decode(newMachine)
+		if err != nil {
+			return err
+		}
+	}
+
+	// we expect to find no nodes within the cluster
+	err := assureNodeCount(0, kubeClient)
```
Comment (p0lyn0mial) [inline test/tools/verify/cmd/main.go]: this is a assumption I would like to have, no nodes within the cluster, what do you think ?

File: test/tools/verify/cmd/main.go [hunk 2]
```diff
@@ -51,19 +57,64 @@ func main() {
...
+		if err == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return fmt.Errorf("falied to created the new machine, err = %v", err)
+	}
+
+	// TODO(lukasz): implement clean-up logic
```
Comment (p0lyn0mial) [inline test/tools/verify/cmd/main.go]: I'd like to clean up the mess we have made. Bonus we could get is that we could validate if our controller knows how to remove unwanted machines.

File: test/tools/verify/cmd/main.go (hunk 1 above)
Comment (mrIncompetent) [reply test/tools/verify/cmd/main.go]: I agree, we should expect 0 nodes within a cluster

File: test/tools/verify/cmd/main.go (hunk 2 above)
Comment (mrIncompetent) [reply test/tools/verify/cmd/main.go]: :+1:

File: test/tools/verify/cmd/main.go [hunk 3]
```diff
@@ -51,19 +57,64 @@ func main() {
...
+	// TODO(lukasz): add dep
```
Comment (p0lyn0mial) [inline test/tools/verify/cmd/main.go]: this is optional, not sure if we want to add dependencies.

Comment (mrIncompetent) [reply test/tools/verify/cmd/main.go]: Could you explain what you mean exactly?
What kind of dependencies?

Comment (p0lyn0mial) [reply test/tools/verify/cmd/main.go]: oh, I was thinking that we could use the `dep` tool. So that we could lock imported(k8s) stuff in.

Comment (mrIncompetent) [reply test/tools/verify/cmd/main.go]: Maybe then we should move the test command to https://github.com/kubermatic/machine-controller/tree/master/cmd ?
Then we would use the same vendor as the controller itself

File: test/tools/verify/cmd/main.go [hunk 4]
```diff
@@ -51,19 +57,64 @@ func main() {
...
+	}
+	err = wait.Poll(500*time.Millisecond, 60*time.Second, func() (bool, error) {
+		err := assureNodeCount(1, kubeClient)
+		if err == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return fmt.Errorf("falied to created the new machine, err = %v", err)
```
Comment (p0lyn0mial) [inline test/tools/verify/cmd/main.go]: at this point we could grab the machine's `Status` from the  kube-api server  and put it into the logs. That would facilitate troubleshooting.

File: test/tools/verify/cmd/main.go (hunk 3 above)
Comment (p0lyn0mial) [reply test/tools/verify/cmd/main.go]: sounds good to me.

**Comment by p0lyn0mial** [comment]
@mrIncompetent is there anything that block this pull? I will follow up clean-up, moving the binary to different directory and printing the `Status` to logs.

File: test/tools/verify/cmd/main.go [hunk 5]
```diff
@@ -51,19 +57,64 @@ func main() {
...
+	if err != nil {
+		return fmt.Errorf("unable to perform the verification, incorrect cluster state detected %v", err)
+	}
+
+	fmt.Printf("creating a new \"%s\" machine\n", newMachine.Name)
+	_, err = machineClient.MachineV1alpha1().Machines().Create(newMachine)
+	if err != nil {
+		return err
+	}
+	err = wait.Poll(500*time.Millisecond, 60*time.Second, func() (bool, error) {
```
Comment (mrIncompetent) [inline test/tools/verify/cmd/main.go]: Should be moved into constants.
Also the timeout is too low. It can take up to 5min

Comment (p0lyn0mial) [reply test/tools/verify/cmd/main.go]: good point.

**Comment by p0lyn0mial** [comment]
@mrIncompetent comments addressed.

**Comment by p0lyn0mial** [comment]
@mrIncompetent just a rebase.

**Pull request merged**